pkg/controller/storagecluster/cephcluster.go

```diff
@@ -47,7 +49,11 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
 	var cephCluster *cephv1.CephCluster
 	// Define a new CephCluster object
 	if sc.Spec.ExternalStorage.Enable {
-		cephCluster = newExternalCephCluster(sc, r.cephImage)
+		monitoringIP, err := r.getExternalMonitoringInfo(sc, reqLogger)
+		if err != nil {
+			return err
+		}
+		cephCluster = newExternalCephCluster(sc, r.cephImage, monitoringIP)
```

Comment:
I see no need to alter the function signature for this. Call `getExternalMonitoringInfo()` from `newExternalCephCluster()`.

Reply:
@jarrpa The reason I did so was that the fuction _newExternalCephCluster_ does not have the _**receiver type** of r *ReconcileStorageCluster_ . Would you rather have that in place of monitoringIP to call getExternalMonitoringInfo() from inside the newExternalCephCluster() ?

Reply:
Ah... shows what I know. :stuck_out_tongue_winking_eye: Okay, I think I can live with this. Let's not break the pattern of the `new*()` functions for this.